cql3/selection/selection.cc

```diff
@@ -598,14 +620,42 @@ void result_set_builder::complete_row() {
     _selectors->add_input_row(*this);
 }
 
+void result_set_builder::accept_new_partition(const std::vector<bytes>& key)
```

Comment:
Per @nuivall : it seems wrong that `accept_new_partition` is triggered for both: new page and new partition events. The code below works around it, but in order not to get ourselves into even bigger troubles, we may want to fix it before addressing groupby vs perpartition limit issue.